pallets/gear/src/manager/journal.rs

```diff
@@ -363,12 +366,30 @@ where
             .unwrap_or_else(|e| unreachable!("Gear bank error: {e:?}"));
     }
 
-    fn store_new_programs(&mut self, code_id: CodeId, candidates: Vec<(MessageId, ProgramId)>) {
+    fn store_new_programs(
+        &mut self,
+        program_id: ProgramId,
+        code_id: CodeId,
+        candidates: Vec<(MessageId, ProgramId)>,
+    ) {
         if let Some(code) = T::CodeStorage::get_code(code_id) {
             let code_info = CodeInfo::from_code(&code_id, &code);
             for (init_message, candidate_id) in candidates {
                 if !Pallet::<T>::program_exists(self.builtins(), candidate_id) {
                     let block_number = Pallet::<T>::block_number();
+
+                    // Make sure an account exists for the newly created program.
+                    // Balance validity check has been performed so we don't expect any errors.
+                    CurrencyOf::<T>::transfer(
+                        &program_id.cast(),
+                        &candidate_id.cast(),
+                        CurrencyOf::<T>::minimum_balance(),
+                        ExistenceRequirement::KeepAlive,
+                    )
+                    .unwrap_or_else(|e| unreachable!("Existential deposit transfer error: {e:?}"));
+                    System::<T>::inc_consumers(&candidate_id.cast())
```

Comment:
lock as well

Reply:
and unlock on exit journal note with the file

Reply:
is there a need if `reducible_balance` is used everywhere?

Reply:
"unlocking" is done in the `clean_inactive_program()` function (called by the exit dispatch handler):
```
// Delete the consumer to allow the account to be reaped.
System::<T>::dec_consumers(&program_account);
```

Reply:
Now literally "unlocking" of the ED is taking place